### ISSUE TITLE
BLT Drush Aliases recipe support for Acquia Cloud Professional

### ISSUE THREAD

**Issue by sebinievas** [acquia/blt#3114]
My system information:
* Operating system type: OSX
* Operating system version: 10.13.6 (17G65)
* BLT version: 9.2.0-alpha3

Output of `blt doctor`:
```
+---------------------------+----------------------------------------------+
| Property                  | Value                                        |
+---------------------------+----------------------------------------------+
| %paths.%root              | /app/docroot                                 |
| %paths.%site              | sites/default                                |
| %paths.%modules           | sites/all/modules                            |
| %paths.%themes            | sites/all/themes                             |
| %paths.%config-sync       | /app/config/default                          |
| %paths.%files             | sites/default/files                          |
| %paths.%temp              | /tmp                                         |
| %paths.%private           | /app/files-private                           |
| admin-theme               | seven                                        |
| alias-searchpaths.0       | /app/drush/sites                             |
| blt-version               | 9.2.0-alpha3                                 |
| bootstrap                 | Successful                                   |
| composer-version          | Composer version 1.7.2 2018-08-16 16:57:12   |
| config-sync               | /app/config/default                          |
| db-driver                 | mysql                                        |
| db-hostname               | database                                     |
| db-name                   | drupal                                       |
| db-password               | drupal                                       |
| db-port                   | 3306                                         |
| db-status                 | Connected                                    |
| db-username               | drupal                                       |
| drupal-settings-file      | sites/default/settings.php                   |
| drupal-version            | 8.6.1                                        |
| drush-cache-directory     | /var/www/.drush/cache                        |
| drush-conf.0              | /app/vendor/drush/drush/drush.yml            |
| drush-conf.1              | /app/drush/drush.yml                         |
| drush-conf.2              | /app/docroot/sites/default/local.drush.yml   |
| drush-script              | /app/vendor/bin/drush                        |
| drush-temp                | /tmp                                         |
| drush-version             | 9.4.0                                        |
| files                     | sites/default/files                          |
| install-profile           | lightning                                    |
| modules                   | sites/all/modules                            |
| php-bin                   | /usr/local/bin/php                           |
| php-conf.1                | false                                        |
| php-os                    | Linux                                        |
| private                   | /app/files-private                           |
| root                      | /app/docroot                                 |
| site                      | sites/default                                |
| stacks.drupal-vm.inited   | false                                        |
| stacks.dev-desktop.inited | false                                        |
| temp                      | /tmp                                         |
| theme                     | bartik                                       |
| themes                    | sites/all/themes                             |
| uri                       | http://local.blted8.com                      |
+---------------------------+----------------------------------------------+
+--------------------------------------+-----------------------------------------------------------+
| Check                                | Problem                                                   |
+--------------------------------------+-----------------------------------------------------------+
| BehatCheck:checkBaseUrl:uri          | base_url in tests/behat/local.yml does not match the site |
|                                      | URI.                                                      |
|                                      |   Behat base_url is set to                                |
|                                      | http://churros.lndo.site.                                 |
|                                      |   Drush site URI is set to                                |
|                                      | http://local.blted8.com.                                  |
| NodeCheck:checkNodeVersionFileExists | Neither .nvmrc nor .node-version file found in repo root. |
| WebUriCheck:checkUriResponse         | Did not get a response from http://local.blted8.com       |
|                                      | Is your *AMP stack running?                               |
|                                      | Is your /etc/hosts file correctly configured?             |
|                                      | Is your web server configured to serve this URI from      |
|                                      | /app/docroot?                                             |
|                                      | Is options.uri set correctly in                           |
|                                      | /app/docroot/sites/default/local.drush.yml?               |
+--------------------------------------+-----------------------------------------------------------+
[error]  BLT Doctor discovered one or more critical issues.  
```

When I run this command:
```
blt recipes:aliases:init:acquia -vvv && drush sa
```

I get the following output:
```
[debug] Drupal VM is not initialized.
Gathering site info from Acquia Cloud.
Gathering sites list from Acquia Cloud.
Found 1 sites for environment dev, writing aliases...
Found 1 sites for environment prod, writing aliases...
Found 1 sites for environment test, writing aliases...
Aliases were written, type 'drush sa' to see them.
3.102s total time elapsed.
 [success] No site aliases found.
```

And I expected this to happen:
Site aliases should be listed. At least three of them for each environment. Also, notice in the `blt doctor` report that the `drush-alias-files` property is not present.

**Comment by lcatlett** [comment]
@sebinievas to help with debugging, can you provide more detail on your drush 9 config by running `drush st --debug -vvv`  in your `/app/` directory and post the output?

**Comment by sebinievas** [comment]
We noticed we were receiving false positives when we were testing BLT against our `Acquia Cloud Professional` accounts. We dug in further and noticed there were some conditional statements where it wasn't checking for `acp`. It checks for `ace` and `acsf` only to populate the `$sites`. Thus condition is not satisfied to write on line `303` of `src/Robo/Commands/Generate/AliasesCommand.php`.

We've gotten around it by changing line `250` with:

```
if ($hosting == 'ace' || $hosting == 'acp') {
```

This works for now, but of course we are a little skeptical, perhaps it needs to be in its own conditional block with some additional checks.

**Comment by sebinievas** [comment]
@lcatlett would you have any suggestions / improvements regarding this issue? Anything welcomed, thanks!

**Comment by lcatlett** [comment]
@sebinievas apologies for the delay -  #3115 has my comments.

**Comment by sebinievas** [comment]
@lcatlett no problemo, I've responded there as well.

**Comment by lcatlett** [comment]
closed in favor of #3115. thanks @sebinievas!
